diff --git a/include/taco/tensor.h b/include/taco/tensor.h
index c462cbd3..d7536dab 100644
--- a/include/taco/tensor.h
+++ b/include/taco/tensor.h
@@ -448,6 +448,11 @@ public:
   /// Set to true to perform the assemble and compute stages simultaneously.
   void setAssembleWhileCompute(bool assembleWhileCompute);
 
+  void setNeedsPack(bool needsPack);
+  void setNeedsCompile(bool needsCompile);
+  void setNeedsAssemble(bool needsAssemble);
+  void setNeedsCompute(bool needsCompute);
+
   /// Get the source code of the kernel functions.
   std::string getSource() const;
 
@@ -510,10 +515,6 @@ private:
   bool neverPacked();
 
   void unsetNeverPacked();
-  void setNeedsPack(bool needsPack);
-  void setNeedsCompile(bool needsCompile);
-  void setNeedsAssemble(bool needsAssemble);
-  void setNeedsCompute(bool needsCompute);
 
   void addDependentTensor(TensorBase& tensor);
   void removeDependentTensor(TensorBase& tensor);
